test/let_done_test.cpp

```diff
@@ -93,3 +93,52 @@ TEST(TransformDone, WithValue) {
   EXPECT_TRUE(multiple.has_value());
   EXPECT_EQ(*multiple, std::tuple(42, 1, 2));
 }
+
+static auto just42() {
+  return just(42);
+}
+
+TEST(TransformDone, WithFunction) {
+  auto freeFunction = just_done() | let_done(just42) | sync_wait();
+
+  EXPECT_TRUE(freeFunction.has_value());
```

Comment:
I like to use `ASSERT_TRUE` for this assertion; if the assertion is ever false, `EXPECT_TRUE` will report the error and continue execution, leading to UB on the next line but `ASSERT_TRUE` will report the error and then end the test (by executing a `return;` statement).
```suggestion
  ASSERT_TRUE(freeFunction.has_value());
```